llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1024,29 +1022,16 @@ static void translateUnPackMask(CallInst &CI) {
 // This function sets VCNamedBarrierCount attribute to set
 // the number of named barriers required by a kernel
 static void translateNbarrierInit(CallInst &CI) {
-  auto *F = CI.getFunction();
-
+  auto F = CI.getParent()->getParent();
```

Comment:
`getFunction` was more concise, IMO

Reply:
It came from translateSLM function. I will change it to 'getFunction' in both funcs.

Reply:
Fixed in e39c9ef6a9abd9d2b94a84c9d32929554da3f841